src/demux.js

```diff
@@ -265,18 +279,67 @@ class DemuxSink extends Writable {
         }
     }
 
+    /**
+     * Validates first if a previously connected sessionId's data has been processed
+     * and checks if there is no missing sequence number
+     *
+     * @param {*} statsSessionId
+     * @param {*} sequenceNumber
+     * @param {*} lastSequenceNumber
+     */
+    _validateSequenceNumber(statsSessionId, sequenceNumber, lastSequenceNumber) {
+        if ((sequenceNumber - lastSequenceNumber > 1)
+        && !fs.existsSync(utils.getDumpPath(this.tempPath, statsSessionId))) {
+            PromCollector.dataIsAlreadyProcessedCount.inc();
+            throw new Error(`[Demux] Session reconnected but file was already processed! sessionId: ${statsSessionId}`);
+        }
+
+        if (sequenceNumber - this.lastSequenceNumber > 1) {
+            this.log.error(`[Demux] sequence number is missing! 
+                sessionId: ${statsSessionId} 
+                sequenceNumber: ${sequenceNumber} 
+                lastSequenceNumber: ${this.lastSequenceNumber}`
+            );
+            PromCollector.missingSequenceNumberCount.inc();
+        }
+    }
+
+    /**
+     *
+     * @param {*} data
+     * @returns {RequestData}
+     */
+    _toRequestData(data) {
+        const jsonData = Array.isArray(data) ? data : JSON.parse(data);
+        const sequenceNumber = jsonData[4];
+        const timestamp = jsonData[3];
+
+        return new RequestData({ timestamp,
+            sequenceNumber });
+    }
+
     /**
      * Handle API requests.
      *
      * @param {Object} request - Request object
      */
     async _handleRequest(request) {
         this._requestPrecondition(request);
-
         PromCollector.requestSizeBytes.observe(sizeof(request));
 
         const { statsSessionId, type, data } = request;
 
+        // save the last sequence number to notify the frontend
+        if (data) {
+            const requestData = this._toRequestData(data);
+
+            this._validateSequenceNumber(statsSessionId, requestData.sequenceNumber, this.lastSequenceNumber);
+
+            this.lastTimestamp = requestData.timestamp;
+            this.lastSequenceNumber = requestData.sequenceNumber;
```

Comment:
in case of the connectionInfo stats entry there is no sequenceNumber, and it might be undefined, please check if the sequenceNumber from the request data is valid